tests/unit/verifiers/test_ovn_central.py

```diff
@@ -0,0 +1,783 @@
+
+    if format_ == "string":
+        return {
+            unit: {
+                "southbound": yaml.dump(status["southbound"], indent=2),
+                "northbound": yaml.dump(status["northbound"], indent=2),
+            }
+            for unit, status in output_data.items()
+        }
+    elif format_ == "ClusterStatus":
+        return {
+            unit: {
+                "southbound": ovn_central.ClusterStatus(
+                    yaml.dump(status["southbound"])
+                ),
+                "northbound": ovn_central.ClusterStatus(
+                    yaml.dump(status["northbound"])
+                ),
+            }
+            for unit, status in output_data.items()
+        }
+    elif format_ == "dict":
+        return output_data
+    else:
+        raise ValueError(f"Unknown output format {format_}")
+
+
+def verify_leader_consistency(
+    expected_result: Result, sample_cluster_data: dict, mocker, model
+):
+    """Verify expected result from consistency check."""
+    # Mock complete_cluster_status property
+    complete_cluster_status = {
+        unit: ovn_central.UnitClusterStatus(**status)
+        for unit, status in sample_cluster_data.items()
+    }
+    cluster_status_property = PropertyMock(return_value=complete_cluster_status)
+
+    mocker.patch.object(
+        ovn_central.OvnCentral,
+        "complete_cluster_status",
+        new_callable=cluster_status_property,
+    )
+
+    # create verifier and run check
+    verifier = ovn_central.OvnCentral([Unit("ovn-central/0", model)])
+
+    result = verifier.check_leader_consistency()
+    assert result == expected_result
+
+
+#  Tests for ovn_central.ClusterStatus class
+def test_cluster_status_init(ovn_cluster_status_raw, ovn_cluster_status_dict):
+    """Test successful initialization of ClusterStatus class from data string."""
+    status = ovn_central.ClusterStatus(ovn_cluster_status_raw)
+    expected_status = ovn_cluster_status_dict
+
+    assert status.cluster_id == expected_status["Cluster ID"]
+    assert status.server_id == expected_status["Server ID"]
+    assert status.status == expected_status["Status"]
+    assert status.role == expected_status["Role"]
+    assert status.vote == expected_status["Vote"]
+    assert status.log == expected_status["Log"]
+    assert status.entries_not_committed == expected_status["Entries not yet committed"]
+    assert status.entries_not_applied == expected_status["Entries not yet applied"]
+    assert status.servers == expected_status["Servers"]
+
+
+def test_cluster_status_init_missing_key(ovn_cluster_status_dict):
+    """Test failure of ClusterStatus initialization when data string is missing key."""
+    missing_key = "Status"
+    ovn_cluster_status_dict.pop(missing_key)
+    status_string = yaml.dump(ovn_cluster_status_dict)
+    with pytest.raises(ovn_central.JujuVerifyError) as exc_info:
+        _ = ovn_central.ClusterStatus(status_string)
+
+    assert (
```

Comment:
Somehow, I don't like to check the string produced by the software unless it is strictly defined.  It is the logic we generate test case "after" programming. If I implement this unit test before programming, I may only check 
1. if the function rasie the exception as I expected? ==> check the exception type
2. if the function throw message with some key word inside, then good enough.  
I hope I still have some freedom to change the string in the code without breaking the unit test.
No good/bad , just my commment. You don't need to change anything here.

Reply:
I see your point. In this case I wanted to make sure that the exception message is properly generated as it's not just static text but also a list of missing attributes.